bids/variables/variables.py

```diff
@@ -448,12 +448,31 @@ def resample(self, sampling_rate, inplace=False, kind='linear'):
         self.index = self._build_entity_index(self.run_info, sampling_rate)
 
         x = np.arange(n)
-        num = int(np.ceil(n * sampling_rate / old_sr))
+        # In Index we trust!
+        num = len(self.index)
+        # _build_entity_index computation above does it per run,
+        # which possibly provides a more stable solution and yadadada
+        num_computed = int(np.ceil(n * sampling_rate / old_sr))
+        num_diff = abs(num - num_computed)
+        if num_diff > 1:
+            raise RuntimeError(
+                "Our internal assumptions about resampling are too faulty "
+                "to continue.")
+        elif num_diff:
+            assert num_diff == 1  # the only way to get here
+            import warnings
+            warnings.warn(
+                "Probably due to multiple runs we ran into a slight "
```

Comment:
This appears with a single run when TR=2.000001. I don't think this is a great error message. It's not clear that it's a warning that will do anything but cause confusion.

Reply:
we added it because there were no other way to inform user (e.g. logging) and we see how silence could be problematic to read down the road ;)  again, with aforementioned suggested fix by just relying on `index` length this would be gone altogether